todo/views.py

```diff
@@ -17,8 +17,35 @@ def createpage(request):
         todo.save()
         return redirect('home')
     return render(request,'create.html')
+
+def delete(request, todo_id):
+    '''delete a todo'''
+    todo=Todo.objects.get(id=todo_id)
+    todo.delete()
+    return redirect('home')
+
+def edit(request, todo_id):
+    '''edit a todo'''
+    todo=Todo.objects.get(id=todo_id)
+    if request.method=='POST':
+        title=request.POST.get('title')
+        desc=request.POST.get('description')
+        todo.title=title
+        todo.description=desc
+        todo.save()
+        return redirect('home')
+    return render(request,'edit.html', {'todo': todo})
+
+def mark_as_done(request, todo_id):
+    '''mark a todo as done'''
+    todo=Todo.objects.get(id=todo_id)
```

Comment:
Same validation as mentioned for delete.

Reply:
Ok